src/RulesFactory.js

```diff
@@ -0,0 +1,69 @@
+import warning from 'warning'
+
+import RegularRule from './rules/RegularRule'
+import SimpleRule from './rules/SimpleRule'
+import KeyframeRule from './rules/KeyframeRule'
+import ConditionalRule from './rules/ConditionalRule'
+import FontFaceRule from './rules/FontFaceRule'
+
+export default class RulesFactory {
+  constructor() {
+    this.classes = {
+      '@charset': SimpleRule,
+      '@import': SimpleRule,
+      '@namespace': SimpleRule,
+      '@keyframes': KeyframeRule,
+      '@media': ConditionalRule,
+      '@supports': ConditionalRule,
+      '@font-face': FontFaceRule
+    }
+    this.regExp = this.buildRegExp()
+  }
+
+  /**
+   * Build regexp for matching rules.
+   *
+   * @return {RegExp}
+   * @api private
+   */
+  buildRegExp() {
+    return new RegExp(`^${Object.keys(this.classes).join('|')}`)
+  }
+
+  /**
+   * Register a new class.
+   *
+   * @param {String} name
+   * @param {Class} cls
+   * @api public
+   */
+  register(name, cls) {
+    this.classes[name] = cls
+    this.regExp = this.buildRegExp()
+  }
+
+  /**
+   * Create a rule instance.
+   *
+   * Options:
+   *   - `className` pass class name if you to define it manually
+   *
+   * @param {Object} [name]
+   * @param {Object} [style] declarations block
+   * @param {Object} [options] rule options
+   * @return {Object} rule
+   * @api public
+   */
+  get(name, style = {}, options = {}) {
```

Comment:
well its actually a classical factory get method, which is supposed to instantiate and return. I know what you mean, I had it this way at first. But then I looked up the factory pattern and they use `get`.